### PR TITLE
Support for DETR model

### LINKED ISSUE #25: [FEATURE] Verify and Integrate DETR to vortex
**Describe the solution you'd like**
Try [DETR](https://github.com/facebookresearch/detr) training with small dataset, if verified, add as a new detection module

also there might be several Vortex limitation for training DETR.
- Additional parameters not covered by existing model parameters in configuration files, may need to add `additional_args` to `model` config
- Forwarding params group to optimizer. Current condition only allow model.parameters() directly forwarded to optimizer. May need additional dedicated model_components key for params gorup so that in the trainer can check whether it exist or not and forward params group to the optimizer if exist rather than whole model.parameters(). Important to be noted that params group modification must be done under create_model_components -> in the model's object side
- validator (`BoundingBoxValidator`) arguments passed to model doesn't have the same behavior as vortex runtime for predictor model (not generalized).

### PR DESCRIPTION
**Pull request by triwahyuu**
## Type of changes

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

closes #25 

## Checklist

- [x] Unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have filled the `[Unreleased]` [Changelog](https://github.com/nodefluxio/vortex/blob/master/CHANGELOG.md)

### CODE REVIEW

File: src/development/vortex/development/utils/data/collater/detr.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import torch
+
+from vortex.development.networks.models.detection.detr import NestedTensor
+from easydict import EasyDict
+
+class DETRColatte:
```
Comment (alphinside) [inline src/development/vortex/development/utils/data/collater/detr.py:6]: should be DETRCollate?

**Comment by alphinside** [comment]
Got error when trying to train using FrontalFDDBDataset with 1 class

```console
  File "/home/nodeflux/.virtualenvs/vortex/lib/python3.6/site-packages/vortex/development/utils/data/collater/detr.py", line 46, in create_collater
    return DETRColatte(*args, **kwargs)
  File "/home/nodeflux/.virtualenvs/vortex/lib/python3.6/site-packages/vortex/development/utils/data/collater/detr.py", line 9, in __init__
    for k,val in dataformat.items()}
  File "/home/nodeflux/.virtualenvs/vortex/lib/python3.6/site-packages/vortex/development/utils/data/collater/detr.py", line 9, in <dictcomp>
    for k,val in dataformat.items()}
AttributeError: 'NoneType' object has no attribute 'items'
```

File: src/development/vortex/development/utils/data/collater/detr.py (hunk 1 above)
Comment (triwahyuu) [reply src/development/vortex/development/utils/data/collater/detr.py:6]: oh yeah, it's a typo

Comment (alphinside) [reply src/development/vortex/development/utils/data/collater/detr.py:6]: Okay, I'll fix it together with the bug below yea

Comment (alphinside) [reply src/development/vortex/development/utils/data/collater/detr.py]: fixed in https://github.com/nodefluxio/vortex/pull/43/commits/55d59038dd4cec7f86f7d85ab8b8be7a2dced7cd

**Comment by triwahyuu** [comment]
okay

**Comment by alphinside** [comment]
LGTM

**Comment by triwahyuu** [comment]
ready to be merged

**Pull request merged**